packages/react/src/createTRPCReact.tsx

```diff
@@ -41,19 +42,42 @@ export type DecorateProcedure<
   TPath extends string,
 > = TProcedure extends AnyQueryProcedure
   ? {
-      useQuery: <
-        TQueryFnData = inferProcedureOutput<TProcedure>,
-        TData = inferProcedureOutput<TProcedure>,
-      >(
-        input: inferProcedureInput<TProcedure>,
-        opts?: UseTRPCQueryOptions<
-          TPath,
-          inferProcedureInput<TProcedure>,
-          TQueryFnData,
-          TData,
-          TRPCClientErrorLike<TProcedure>
-        >,
-      ) => UseTRPCQueryResult<TData, TRPCClientErrorLike<TProcedure>>;
+      useQuery:
+        | (<
+            TQueryFnData = inferProcedureOutput<TProcedure>,
+            TData = inferProcedureOutput<TProcedure>,
+          >(
+            input: inferProcedureInput<TProcedure>,
+            opts?: Omit<
+              UseTRPCQueryOptions<
+                TPath,
+                inferProcedureInput<TProcedure>,
+                TQueryFnData,
+                TData,
+                TRPCClientErrorLike<TProcedure>
+              >,
+              'initialData'
+            > & { initialData?: () => undefined },
+          ) => UseTRPCQueryResult<TData, TRPCClientErrorLike<TProcedure>>)
+        | (<
```

Comment:
For overloading a function type you need to use `&` instead of `|`. Also, I think the`opts` param cannot be optional for this overload.